core/src/databases/table_schema.rs

```diff
@@ -262,6 +265,85 @@ impl TableSchema {
                 .collect::<Result<Vec<_>>>(),
         }
     }
+
+    pub fn merge(&self, other: &Self) -> Result<Self, anyhow::Error> {
+        // Index our and other columns by name.
+        let our_columns_set = self.0.iter().map(|c| &c.name).collect::<HashSet<_>>();
```

Comment:
This is mall and used only once per new other_columns. I'd be in favour of removing that variable and use a find instead.

Reply:
Makes the code harder to understand yet looping on a 10 value array is instantaneous. So that's a tradeoff beftween readibility and performance and I find the tradeoff clearly on one side here. But fine with keeping as is of course.

Reply:
I kind disagree that an inline find is much more readable than a set.contains. In fact, I actually think the latter is more readable

Reply:
Using an algorithm with sub-optimal complexity is more idiomatic in rust ?